### ISSUE TITLE
Current Dune alias syntax can't work reliably on Windows

### ISSUE THREAD

**Issue by jonahbeckford** [ocaml/dune#5775]
## Desired Behavior

I maintain the [Diskuv OCaml distribution](https://diskuv.gitlab.io/diskuv-ocaml), and will be soon releasing a shim for Dune so that Windows users can just type `dune build` to build like everybody else does in *nix. The shim will bring a MSYS environment for any Unix binaries that a Dune project needs and will also bring MSVC so OCaml can do native compilation.

Unfortunately [Dune aliases](https://dune.readthedocs.io/en/latest/usage.html#aliases) use the `@` symbol. That means every Windows user who uses PowerShell (the default Windows shell in popular programs like Windows Terminal and VS Code) will be forced to escape any aliases. It sounds like a small problem, but it is a problem that will become a major complaint as more people use Windows on OCaml. So instead of:

```powershell
dune build @foo
```

Windows users have to copy, paste **and modify** it to:

```powershell
dune build `@foo
```

If they don't, the `@` term can be silently removed because it is the PowerShell array operator. Have a look at:

```powershell
$ echo dune build @foo
dune
build

$ echo dune build @foo/src/all
dune
build
@foo/src/all
```

Basically, the `@foo` array in PowerShell is acting like the `${foo[@]}` array in Bash ... if the array variable does not exist it is empty.

Hopefully it is apparent how confusing or frustrating that will be to Windows Dune users, and time-consuming for any OCaml package maintainer who has to respond to bug reports why their `dune` command line does not work.

## Proposed

Either of the following:

1. Use `dune build =foo` or `dune build +foo` instead of `dune build @foo`.
2. Use `dune alias foo` instead of `dune build @foo`.

In both cases I would remove `dune build @foo` from your documentation but keep that syntax around for backwards compatibility.

**Comment by nojb** [comment]
I think this makes sense.

**Comment by nojb** [comment]
We discussed this during the dev meeting. The following possibilities were considered:

1. Adding an alternative character instead of `@` (we can not really remove `@` in the short term because it is widely used)
2. Adding a dedicated command, eg `dune build alias` or `dune alias build` or `dune alias`. Downside: it is not composable, in the sense that one cannot build both aliases and non-aliases in the same command line.
3. Add a flag to `dune build`, eg `dune build --alias foo bar` would build the alias `foo` and the target `bar`.

We also mentioned a little-known feature of Dune which makes it possible to build *any* dependency specification from the command line, eg `dune build '(alias foo)'`, but of course this syntax is not useful for day-to-day use.

1. is possible, but am not sure adding yet another random "magical" character is the best way forward. 3. may be a better way forward.

**Comment by jonahbeckford** [comment]
Thanks. IMHO option 3 seems best (from a beginner perspective it is the easiest syntax to understand), but any of the options works.

**Comment by yawaramin** [comment]
Another option could be using a prefix eg `dune build alias:foo`. dune could parse `alias:XYZ` equivalent to `@XYZ`.

**Comment by Heyji2** [comment]
If I may, I would suggest an informative message in case the default target is chosen, as it was not obvious for me that the default target was chosen instead of the one I had provided. This would have been a good hint though about why the command is not behaving as expected.  Wether this message should be provided all the time or only when the --verbose option is supplied is up to you. This can be implemented in addition to any choice suggested above.

**Comment by yawaramin** [comment]
Need to think carefully about what would be in that informative message. If it says eg `Building target @all`, users could get confused as to why it's building `@all` when they asked for `@doc`.

**Comment by Alizter** [comment]
In #12043 the arguments `--alias` and `--alias-rec` were added allowing an alternative way of specifying aliases to `dune build`.
